idempotence4j-spring-boot-starter/src/main/java/com/transferwise/idempotence4j/autoconfigure/RetentionProperties.java

```diff
@@ -6,10 +6,18 @@
 @Data
 public class RetentionProperties {
     /**
-     * Retention period specified in a ISO-8601 format. Can only be positive
+     * Retention period specified in a ISO-8601 format. Can only be positive.
+     * Either period or duration must be filled.
      * Default value: 30 days
      */
     private String period = "P0Y0M30D";
+
+    /**
+     * Retention duration specified in ISO-6801 format. Can only be positive.
+     * Either period or duration must be filled.
+     * Defaults to null.
+     */
+    private String duration = null;
```

Comment:
you could use `Duration` class here directly https://docs.spring.io/spring-boot/docs/2.1.13.RELEASE/reference/html/boot-features-external-config.html#boot-features-external-config-conversion-duration